test/testing/test_services.py

```diff
@@ -205,6 +205,28 @@ class BadService(Service):
             call()
 
 
+def test_entrypoint_hook_timeout(container_factory, rabbit_config):
+
+    class Service:
+        name = 'service'
+
+        @rpc
+        def long_task(self):
+            time.sleep(0.1)
+            return
```

Comment:
CI is failing because this line is not covered. The return is not required though, so you can just delete this line.

Reply:
Thanks for spotting that.

I'm having some problems with running the whole test suite using `make test`. I'm using Python 3.6.8, doing the `pip install -e .[devs]` and running rabbitmq in Docker binding 5672 and 15672 ports to the host. It's failing on some tests in `test_events.py` and even freezes on one.

When I cancel it (using Ctrl+C) then I get
`Error connecting to broker at pyamqp://guest:********@localhost:5671/nameko_test_tdgwoaqweb ([Errno 111] ECONNREFUSED)`.

The example failing test assertions are like:
```
____________________________________________________ test_proxy_disconnect_with_active_worker ____________________________________________________

container_factory = <function container_factory.<locals>.make_container at 0x7f96045868c8>
rabbit_manager = <nameko.testing.rabbit.Client object at 0x7f9604575400>
rabbit_config = {'AMQP_URI': 'pyamqp://guest:guest@localhost:5672/nameko_test_mvudekhpwp', 'username': 'guest', 'vhost': 'nameko_test_mvudekhpwp'}

    def test_proxy_disconnect_with_active_worker(
            container_factory, rabbit_manager, rabbit_config):
        """ Break the connection to rabbit while a service's queue consumer and
        rabbit while the service has an in-flight rpc request (i.e. it is waiting
        on a reply).
        """
        # ExampleService is the target; ProxyService has the rpc_proxy;
        proxy_container = container_factory(ProxyService, rabbit_config)
        example_container = container_factory(ExampleService, rabbit_config)

        proxy_container.start()

        # get proxyservice's queue consumer connection while we know it's the
        # only active connection
        vhost = rabbit_config['vhost']
        connections = get_rabbit_connections(vhost, rabbit_manager)
>       assert len(connections) == 1
E       assert 0 == 1
E        +  where 0 = len([])

test/test_broker.py:116: AssertionError
------------------------------------------------------------- Captured log teardown --------------------------------------------------------------
mixins.py                  180 WARNING  Connection to broker lost, trying to re-establish connection...
Traceback (most recent call last):
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/kombu/mixins.py", line 175, in run
    for _ in self.consume(limit=None, **kwargs):
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/kombu/mixins.py", line 197, in consume
    conn.drain_events(timeout=safety_interval)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/kombu/connection.py", line 315, in drain_events
    return self.transport.drain_events(self.connection, **kwargs)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/kombu/transport/pyamqp.py", line 103, in drain_events
    return connection.drain_events(**kwargs)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/connection.py", line 500, in drain_events
    while not self.blocking_read(timeout):
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/connection.py", line 506, in blocking_read
    return self.on_inbound_frame(frame)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/method_framing.py", line 55, in on_frame
    callback(channel, method_sig, buf, None)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/connection.py", line 510, in on_inbound_method
    method_sig, payload, content,
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/abstract_channel.py", line 126, in dispatch_method
    listener(*args)
  File "/home/damian/.pyenv/versions/3.6.8/envs/nameko/lib/python3.6/site-packages/amqp/channel.py", line 1458, in _on_basic_cancel
    raise ConsumerCancelled(consumer_tag, spec.Basic.Cancel)
amqp.exceptions.ConsumerCancelled: Basic.cancel: (0) None4
```